### ISSUE TITLE
Odometer in 1.58 beta ?

### ISSUE THREAD

**Issue by pimisen** [itchannel/fordpass-ha#370]
I see you changed something in the way odometer is filled with value.
But for me it is now completely missing as an entity.
European Car (S-Max 2020) -> Metric system
sensor.fordpass_outsidetemp is also missing

**Comment by tcc0** [comment]
Fwiw; still works for me with 1.58 beta 1 (Focus MK4)

**Comment by serlinGi** [comment]
Hello and thank you for your countless efforts these last few days.  
I  have integrated 2 cars: 1 Kuga and 1 EcoSport;  with version 1.57 the sensor values were all valorised (even if I don't check if all were correct);  

It's also true that after a few hours of operation they disconnected but I had modified the fordpass_new.py file a few hours ago as per #360 and for the moment it was still OK.  

Now with 1.58be5a1 for the kuga the sensor.fordpass_odometer is unavailable and is OK for EcoSport;  sensor.fordpass_outsidetemp is unavailable for Kuga and is not present with EcoSport.  
Upon restart, in the LOG I receive a Warning for the first: "Entity sensor.fordpass_odometer_2 (<class 'custom_components.fordpass.sensor.CarSensor'>) is using native unit of measurement 'None' which is not a valid unit for the device class ( 'distance') it is using; expected one of ['in', 'yd', 'ft', 'km', 'cm', 'mi', 'm', 'mm']; ..." ;  

for the other sensor "Error adding entities for domain sensor with platform fordpass...... Error "File "/config/custom_components/fordpass/sensor.py", line 283, in get_value
     return {"Ambient Temp": self.data["ambientTemp"]["value"]}
 KeyError: 'ambientTemp'.  "  

If I can do some steps to help you with the checks, tell me, but consider that you should explain the ABCs to me: I'm really poor😅

Thanks again

**Comment by SquidBytes** [comment]
I am also going to guess that
engineCoolantTemp
outsideTemperature
engineOilTemp
values are all off, and not accurate.

1.58 beta moved over to using HA as conversions.
I had previously added conversions in the code.
These code conversions were working, and then HA was also converting them.
This also broke the ambientTemp attribute I added under outsideTemp here

> "Error adding entities for domain sensor with platform fordpass...... Error "File "/config/custom_components/fordpass/sensor.py", line 283, in get_value
return {"Ambient Temp": self.data["ambientTemp"]["value"]}
KeyError: 'ambientTemp'. "

Because that line broke, other sensors were not getting updates and that should explain why some show up as unavailable.
This also should be @pimisen issue, just showing slightly different because of the metric vs imperial.

I believe I have fixed this, and added it to my PR.
If you want to test, you can replace your `sensor.py` file within HomeAssistant with this `sensor.py` here: https://github.com/SquidBytes/fordpass-ha/blob/1.58/custom_components/fordpass/sensor.py
While the PR is awaiting approval.

For this error:
> "Entity sensor.fordpass_odometer_2 (<class 'custom_components.fordpass.sensor.CarSensor'>) is using native unit of measurement 'None' which is not a valid unit for the device class ( 'distance') it is using; expected one of ['in', 'yd', 'ft', 'km', 'cm', 'mi', 'm', 'mm']; ..." ;

It looks like you have made a custom sensor `sensor.fordpass_odometer_2`
Because HA is now using conversion, whatever you had mapped to that custom sensor is not a measurement.

**Comment by itchannel** [comment]
Easy fixes. 

For the odometer issue this is the offending line 

```
            if self.sensor == "odometer":
                if self.fordoptions[CONF_DISTANCE_UNIT] == "mi":
                    return "km"
```

It should now just be 
```
            if self.sensor == "odometer":
                return "km"
```

Ambient Temp can be fixed changing:
```
            if self.sensor == "outsideTemp":
                if self.fordoptions[CONF_DISTANCE_UNIT] == "mi":
                    return {"Ambient Temp": round(float(self.data["ambientTemp"]["value"] * 9/5) + 32)}
                return {"Ambient Temp": self.data["ambientTemp"]["value"]}
```

to

```
            if self.sensor == "outsideTemp":
                 if "ambientTemp" in self.data:  
                    return {"Ambient Temp": self.data["ambientTemp"]["value"]}
```

I will check @SquidBytes pull request later to see if it fixes this or I'll push a change myself later :)

**Comment by serlinGi** [comment]
Thx @SquidBytes !

I just uploaded the new sensor.py and I will let you know soon.  

Regarding odometer_2, it is not a custom sensor but I have
2 cars: all entities in the second one are named with _2 at the end.

**Comment by serlinGi** [comment]
OK, done and no warnings or errors; 👍
now the sensors are all available (except deepsleep in progress and firmwareupg in progress which have never been available before); 
the only note which is very strange: the 2 cars have been stopped since this morning and the coolant temp on the EcoSport is 28 °C - correct - while on the Kuga it is a good 91 °C  😬 ; also the engineOilTemp is 79°C...  😰😂

**Comment by SquidBytes** [comment]
> Thx @SquidBytes !
> 
> I just uploaded the new sensor.py and I will let you know soon.
> 
> Regarding odometer_2, it is not a custom sensor but I have 2 cars: all entities in the second one are named with _2 at the end.

That makes sense. That didn't cross my mind lol.
Still new to Hass and doing this.

**Comment by nytram-md** [comment]
Hi

Ford Focus 2021 Diesel

Ive just updated from 157 to 158 beta and lost fordpass_odometer and fordpass_outside_temp

I seem to have lost fordpass_gps in all this the last few days since the API changed. Has that gone now 

Is there a way to get a friendly name again for the sensors too as all mine are showing fordpass_ ?

Thanks again for all your help

**Comment by itchannel** [comment]
Fixed in 1.58 just released.
